llama_index/indices/vector_store/marqo_index.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Sequence
+
+from llama_index.data_structs.data_structs import IndexDict
+from llama_index.schema import BaseNode, ImageNode, IndexNode
+from llama_index.vector_stores.types import NodeWithEmbedding
+from llama_index.indices.vector_store.base import VectorStoreIndex
+from llama_index.vector_stores.marqo import MarqoVectorStore
+
+
+class MarqoVectorStoreIndex(VectorStoreIndex):

```

Comment:
Right now, the preferred approach is to use the same generic vector store index for all vector stores.

Is there a reason we need a specific index for Marqo?

Reply:
Currently it is unused in any of the examples, except for one test. Since marqo doesn't support direct insertion of vectors, I made a custom _add_nodes_to_index function that is able to add nodes into marqo, in case it is ever needed.

However in the example notebooks I used VectorStoreIndex with the MarqoVectorStore. We can delete this class if it is unnecessary.

Reply:
Actually, it looks like the `is_embedding_query` property controls this? 🤔

Reply:
Looks like we can use `is_embedding_query` flag?

Reply:
Thanks for the comment, would setting `is_embedding_query` in the definition of `class MarqoVectorStore(VectorStore):` be sufficient to avoid the computation of embeddings before insertion into marqo?